### DISCUSSION TITLE
script and updateScript, what is the right syntax?

### DISCUSSION THREAD

**Discussion by wyxcoder** [GrapesJS/grapesjs#5186] [General]
I see in document, you can add script to model like

```
editor.Components.addType('comp-with-js', {
  model: {
    defaults: {
      script,
      // Define default values for your custom properties
      myprop1: 'value1',
      myprop2: '10',
```

From other example, you can call updateScript and define script like this
```
editor.Components.addType('comp-with-js', {
  model: {
    defaults: {
       ....
   script: function() {
   .....

  this.listenTo(this, 'change:item_count', this.updateScript);

```

What is the difference? I tested and they work in different way, any idea?

**Comment by artf** [comment]
`script` are functions in both cases, in the first example it's simply defined on top and assigned inside.
And as described [here](https://grapesjs.com/docs/modules/Components-js.html#passing-properties-to-scripts), if you need to pass component properties to the `script` function you have to use `script-props` and you don't need to setup any listener as the script will be updated automatically
